sharktank/sharktank/ops/default_impls.py

```diff
@@ -25,13 +25,33 @@
 from ._registry import AllOfType, AllOfExprs, AllOfExprsVariadic, IsOfType
 from .signatures import *
 import iree.turbine.ops.iree
+from iree.turbine.aot import DeviceTensorTrait
 
 
 @cat.override(AllOfType(Tensor, PrimitiveTensor))
 def cat_default(tensors: Sequence[Tensor | PrimitiveTensor], dim: int):
     return torch.cat([unbox_tensor(t) for t in tensors], dim)
 
 
+@clone.override(IsOfType(InferenceTensor))
+def clone_default(tensor: InferenceTensor):
+    return tensor.clone()
+
+
+@clone.override(IsOfType(Tensor))
+def clone_torch(tensor: Tensor):
+    # Clone introduces a tensor copy operation in the graph. It does not make sense to
+    # clone also the ExternalTensorTrait. The expectation is that when a user clones a
+    # tensor if they modify in-place the result this should not affect the original
+    # tensor. If we are to clone also the ExternalTensorTrait then we would have
+    # external_name aliasing, but the expectation is that there is no data aliasing.
```

Comment:
For pipeline parallelism I need the external tensor trait behaviour to copy. We don't want to bake the pipeline parallelism decisions into the weights, so they are applied to the tensors when loaded from file inside an export script, e.g. `export_paged_llama_v1.py`, FYI changes aren't up yet.

.clone was added for the following behaviour:
1. Load tensor t.
2. Change .devices or move .shards to different devices with ops.transfer_to_logical_device
3. t_new = t.clone(ts=moved_shards, devices=new_devices).
4. Store back into theta.
5. Continue exporting model.

Reply:
If pipeline stages are not going to be a part of the same compute graph then you could do a deepcopy.

Reply:
I think that there is a discrepancy of expectations.

1. Usually a `clone` method does a deep copy.
2. In torch this term is loaded with more meaning as it creates a tensor that is a part of the compute graph. I think we should avoid deviating from that meaning as it will introduce confusion down the line.  If operations under `sharktank.ops` or methods of tensors have the same name as in torch then they should do the same.

Reply:
Perhaps the name is the problem then. I did not introduce this to be equivalent to torch.clone.

Reply:
Yes, another name then would be desirable as we may want something like `torch.clone` as some point.